### PR TITLE
XDP fixes

### PR DESCRIPTION
**Pull request by jacobvw**
Fix eBPF program being rejected by kernel versions < 5.3
Improve XDP statistics counting.
Add XDP to live tests
Fix issues when running XDP on a parallel trace with a dedicated hasher thread
Add TRACE_OPTION_XDP_HARDWARE_OFFLOAD config option
Various other updates and bug fixes

### CODE REVIEW

File: lib/format_linux_xdp.c [hunk 1]
```diff
@@ -618,13 +643,15 @@ static int linux_xdp_pstart_input(libtrace_t *libtrace) {
...
-    if (FORMAT_DATA->state == XDP_PAUSED) {
-        /* update state and return */
-        linux_xdp_set_control_map_state(libtrace, XDP_RUNNING);
-        FORMAT_DATA->state = XDP_RUNNING;
-
-        return 0;
+    switch (FORMAT_DATA->state) {
+        case XDP_PAUSED:
+            /* update state and return */
+            linux_xdp_update_state(libtrace, XDP_RUNNING);
```
Comment (rsanger) [inline lib/format_linux_xdp.c:649]: Should this be XDP_PAUSED rather than XDP_RUNNING?

Comment (rsanger) [reply lib/format_linux_xdp.c:649]: I see that is clearly correct as is now.

**Comment by rsanger** [comment]
Looks good to me, I've have tested with Debian Buster (Linux 4.19) and Ubuntu 20.04 (Linux 5.4).

My only comment is that maybe it is worth waiting and retrying once or twice on an EBUSY return when setting up the XDP socket (linux_xdp_start_stream). Given running XDP in short succession causes problems. It would simplify the tests and we could remove the extra sleep, and it would also useful for anyone using a watchdog to restart a libtrace program.

**Comment by jacobvw** [comment]
> Looks good to me, I've have tested with Debian Buster (Linux 4.19) and Ubuntu 20.04 (Linux 5.4).
> 
> My only comment is that maybe it is worth waiting and retrying once or twice on an EBUSY return when setting up the XDP socket (linux_xdp_start_stream). Given running XDP in short succession causes problems. It would simplify the tests and we could remove the extra sleep, and it would also useful for anyone using a watchdog to restart a libtrace program.

Cool thanks for testing. I have added a retry when EBUSY is returned and removed the delay in the test script.

**Pull request merged**